File: solutions/erlang/anagram/1/src/anagram.erl
-module(anagram).

-export([find_anagrams/2]).


find_anagrams(Subject, Candidates) -> 
	[ Candidate || Candidate <- Candidates, 
								 is_anagram(string:lowercase(Subject), string:lowercase(Candidate)) ].


is_anagram(Word, Word) -> false;
is_anagram(Word, Subject) -> is_anagram0(Word, Subject).

is_anagram0([], []) -> true;
is_anagram0([], _) -> false;
is_anagram0(_, []) -> false;
is_anagram0([FirstLetter | Rest], Candidate) -> 
	lists:member(FirstLetter, Candidate) andalso 
	is_anagram0(Rest, lists:delete(FirstLetter, Candidate)).

File: solutions/python/anagram/1/anagram.py
def find_anagrams(word, candidates):
    low_word = word.lower()
    sorted_word = sorted(low_word)
    return [
        candidate 
        for candidate in candidates 
        if sorted(low_cand := candidate.lower()) == sorted_word 
        and low_cand != low_word]
